### PR TITLE
Adding phpipam_inventory.py script

### PR DESCRIPTION
**Pull request by cidrbl0ck**
Phpipam custom inventory script, couldn't find any examples anywhere for this source. Want to give back.

This was written with AAP usage in mind but I believe it'll be fine as-is for AWX and Tower. Relies upon a custom credential containing the Phpipam API: appid, token value, and query_url.

### CODE REVIEW

File: inventory/phpipam_inventory.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    "token": api_key,
+    "Content-Type": "application/json"
+}
+
+response = requests.get(url, headers=headers)
+print("*********************************************")
+# Uncomment the below two lines to help when troublshooting to dump the contents of the API call to the console.
+
+#response_dict = response.json()
+#print(json.dumps(response_dict, indent=4))
```
Comment (felixfontein) [inline inventory/phpipam_inventory.py:54]: Can you add spaces between `#` and the first letter of the comments?

File: inventory/phpipam_inventory.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# the app_id is the 'name' of your API key
+app_id = os.environ.get("IPAM_ID")
+controller = "addresses"
+action = "get"
+filter_by = "custom_fields"
+custom_field = "custom_managed"
+filter_value = "Yes"
+# Get the inventory file path from an environment variable, used when outputting inventory to
+# a static file.
+#inventory_file_path = os.environ.get('IPAMDEV_FILEPATH')
```
Comment (felixfontein) [inline inventory/phpipam_inventory.py:32]: Same here.

**Comment by felixfontein** [comment]
Please ignore the failing CI for now... I re-activated it after it hadn't run for a looong time... And surprise, it doesn't work anymore ;)

**Comment by felixfontein** [comment]
CI will be fixed by #12.

**Comment by felixfontein** [comment]
Restarting CI.

File: inventory/phpipam_inventory.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    custom_role = address["custom_fields"].get("custom_role", None)
+    custom_managed = address["custom_fields"].get("custom_managed", None)
+
+    if hostname is None or hostname.strip() == "":
+        hostname = ip_address
+
+    if custom_role is not None and custom_role.strip() != "":
+       if custom_role not in custom_roles:
+            custom_roles[custom_role] = {"hosts": {}}
+        custom_roles[custom_role]["hosts"][hostname] = {"ansible_host": ip_address}
```
Comment (felixfontein) [inline inventory/phpipam_inventory.py:77]: That should fix the failing tests:
```suggestion
       custom_roles[custom_role]["hosts"][hostname] = {"ansible_host": ip_address}
```

**Comment by cidrbl0ck** [comment]
Fixed the indentations mentioned.

**Comment by felixfontein** [comment]
The remaining issues should get fixed by #13.

**Comment by felixfontein** [comment]
And everything passes. Finally :)

**Comment by felixfontein** [comment]
@cidrbl0ck thanks for your contribution!

**Pull request merged**